File: parser/src/test/java/com/aaroncoplan/waterfall/parser/tests/ForBlockTests.java
package com.aaroncoplan.waterfall.parser.tests;

import org.junit.Test;

public class ForBlockTests {

    private final String template = "module m {\nfunc f() {\nmyList := [1,2,3]\n%s\n}\n}";

    @Test
    public void testForInEmptyBlock() {
        TestUtils.shouldPass(String.format(template, "for (x in myList) {}"));
    }

    @Test
    public void testForInWithStatements() {
        TestUtils.shouldPass(String.format(template, "for (x in myList){\nprint(x)\n}"));        
    }
}

File: parser/src/test/kotlin/com/aaroncoplan/waterfall/parser/tests/ForBlockTests.kt
package com.aaroncoplan.waterfall.parser.tests

import org.junit.Test

class ForBlockTests {

    private val template = "module m {\nfunc f() {\nmyList := [1,2,3]\n%s\n}\n}"

    @Test
    fun testForInEmptyBlock() {
        TestUtils.shouldPass(template.format("for (x in myList) {}"))
    }

    @Test
    fun testForInWithStatements() {
        TestUtils.shouldPass(template.format("for (x in myList){\nprint(x)\n}"))
    }
}
